src/store/savings/epics.ts

```diff
@@ -58,31 +59,42 @@ const loadAllSavingsItemsAndStakes: Epic = (action$: Observable<Action>, state$:
   action$.pipe(
     ofType(loadAllSavingsAndStakesAction),
     withUsdToTokenRates(state$),
-    switchMap(([, rates]) => getYouvesSavingsItems$(rates)),
+    switchMap(([, rates]) => forkJoin([getYouvesSavingsItems$(rates), getKordFiItems$(rates)])),
     withSelectedAccount(state$),
-    switchMap(([savings, selectedAccount]) => {
-      if (savings.length === 0) {
+    switchMap(([[youvesSavings, kordFiSavings], selectedAccount]) => {
+      if (youvesSavings.length === 0 && kordFiSavings.length === 0) {
         throw new Error('Failed to fetch any savings items');
       }
 
-      return forkJoin(
-        savings.map(savingsItem =>
-          getUserStake(selectedAccount, savingsItem.id, savingsItem.type)
-            .then((stake): [string, UserStakeValueInterface | undefined] => [savingsItem.contractAddress, stake])
-            .catch(e => {
-              console.error('Error while loading farm stakes: ', savingsItem.contractAddress);
-              showStakeLoadError(e);
+      return forkJoin([
+        forkJoin(
+          youvesSavings.map(savingsItem =>
+            getUserStake(selectedAccount, savingsItem.id, savingsItem.type)
+              .then((stake): [string, UserStakeValueInterface | undefined] => [savingsItem.contractAddress, stake])
+              .catch(e => {
+                console.error('Error while loading farm stakes: ', savingsItem.contractAddress);
+                showStakeLoadError(e);
 
-              return [savingsItem.contractAddress, undefined];
-            })
-        )
-      ).pipe(
-        map(stakesEntries =>
-          Object.fromEntries(
-            stakesEntries.filter((entry): entry is [string, UserStakeValueInterface] => isDefined(entry[1]))
+                return [savingsItem.contractAddress, undefined];
+              })
           )
         ),
-        mergeMap(stakes => merge(of(loadAllSavingsActions.success(savings)), of(loadAllStakesActions.success(stakes))))
+        getKordFiUserDeposits$(selectedAccount.publicKeyHash)
+      ]).pipe(
+        map(([youvesStakesEntries, kordFiStakesEntries]) => {
+          return {
+            ...Object.fromEntries(
+              youvesStakesEntries.filter((entry): entry is [string, UserStakeValueInterface] => isDefined(entry[1]))
+            ),
+            ...kordFiStakesEntries
+          };
+        }),
```

Comment:
```suggestion
        map(([youvesStakesEntries, kordFiStakesEntries]) => ({
          ...Object.fromEntries(
            youvesStakesEntries.filter((entry): entry is [string, UserStakeValueInterface] => isDefined(entry[1]))
          ),
          ...kordFiStakesEntries
        })),
```
If you think that it's less convenient, you may resolve this discussion silently